curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-algorithmic-thinking-by-building-a-number-sorter/641110e4fb696b259dbf0bcf.md

```diff
@@ -0,0 +1,384 @@
+---
+id: 641110e4fb696b259dbf0bcf
+title: Step 34
+challengeType: 0
+dashedName: step-34
+---
+
+# --description--
+
+For this algorithm, you'll want to use a `while` loop. This loop needs two conditions:
+
+- First, it should not run beyond the beginning of the array (accessed with `j`).
+- Second, it should not run after it finds a value smaller than the current value.
```

Comment:
Bit confusing what the second "it" is referring to.

Reply:
```suggestion
- Second, the loop should not run after it finds a value smaller than the current value.
```

Better?

Reply:
I was referring to the second 'it" which is the nested loop?

Reply:
Sound good!